src/core/message.c

```diff
@@ -427,6 +432,17 @@ nni_msg_dup(nni_msg **dup, const nni_msg *src)
 	nni_atomic_init(&m->m_refcnt);
 	nni_atomic_set(&m->m_refcnt, 1);
 
+	// clone protocol data if a method was supplied.
+	if (src->m_proto_ops != NULL && src->m_proto_ops->msg_free != NULL) {
```

Comment:
My instinct is that this should be per option, rather than per protocol.  See my comments above.

Reply:
This is from your code.... I was having an effort to persuade you to do so, perhaps we need another talk.

Reply:
Possibly.  I retain the prerogative to change my mind as I think more about these problems. :-)